beacon-chain/sync/service.go

```diff
@@ -252,7 +252,12 @@ func (s *Service) registerHandlers() {
 					return
 				}
 				// Register respective pubsub handlers at state synced event.
-				s.registerSubscribers()
+				digest, err := s.currentForkDigest()
+				if err != nil {
+					log.WithError(err).Error("Could not retrieve current fork digest")
+				}
```

Comment:
There should be a `return` here, right? Because you later call `s.registerSubscribers(currentEpoch, digest)` which does not make sense when `digest` could not be fetched properly.